doc/api/colors_api.rst

```diff
@@ -11,39 +11,157 @@ The Color :ref:`tutorials <tutorials-colors>` and :ref:`examples
    :no-members:
    :no-inherited-members:
 
-Classes
--------
+Color Conversion tools
+----------------------
+
+.. autosummary::
+   :toctree: _as_gen/
+   :template: autosummary.rst
+
+   is_color_like
+   same_color
+
+   hsv_to_rgb
+   rgb_to_hsv
+
+   to_hex
+   to_rgb
+   to_rgba
+   to_rgba_array
+
+   get_named_colors_mapping
+
+
+.. _norms_and_colormaps:
+
+Normalization and Colormapping
+------------------------------
+
+Some `~.artists.Artist` classes can map an array of input data to RGBA
+values, (ex `~.axes.Axes.scatter` or `~.axes.Axes.imshow`).  The
+machinery for this is implemented via the `~.cm.ScalarMappable` base
+class in `~.cm` and the `~.Normalize` and `~.Colormap` classes in
+`~.colors` (this module).
+
+At the core, colormapping is going from a scalar value to a RGB tuple
```

Comment:
I still feel this is more complicated than it needs to be, and not even particularly true if you want to be mathematically rigorous; colormaps are finite discrete ordered points in R^3, not a smooth curve or path, so I find taking about it like this distracting.   The first step is normalizing the data between vmin and vmax to the range 0-1, perhaps using a non-linear function, and secondly, using a linear lookup table to find the RGBA color closest to that value between 0 and 1.

Reply:
I always like code examples to make the abstract concrete:
```
A color mapping from floating point data values to colors takes place in two steps.  First the data is normalized between 0 and 1.  This requires a `vmin` and `vmax` the define which values are mapped to 0 and 1, respectively, and it requires a function that defines the mapping between these two values.    The simplest case is a linear map ::
    
		import matplotlib.colors as mcolors
		norm = mcolors.Normalize(vmin=100, vmax=300) 
		norm([100, 200, 300])  # returns 0.0, 0.5, 1.0

but a logarithmic norm is also possible ::

    		norm = mcolors.LogNorm(vmin=10, vmax=1000)
		norm([10, 100, 1000])  # returns 0.0, 0.5, 1.0

Once the data has been normalized between 0 and 1, it can be passed to a colormap to return RGB(A) values ::
		
		cmap = cm.viridis
		cmap([0.0, 0.5, 1.0])  
	      #  returns RGBA array with the first three colors of viridis colormap.
	      # array([[0.267004, 0.004874, 0.329415, 1.      ],
             # [0.127568, 0.566949, 0.550556, 1.      ],
             # [0.993248, 0.906157, 0.143936, 1.      ]])
```

Reply:
We use discrete colormaps as an implementation detail (both for performance and because we are targeting 8bit RGB which is inherently discrete), but if we had infinite precision floats and an analytic function for the colormap the whole system would still hold together.

I think the correct solution is to include both versions :)